### ISSUE TITLE
SingleVariable objective should be compatible with QPs

### ISSUE THREAD

**Issue by matbesancon** [jump-dev/DiffOpt.jl#82]
```julia
const QP_OBJECTIVE_TYPES = Union{
    MathOptInterface.ScalarAffineFunction{Float64},
    MathOptInterface.ScalarQuadraticFunction{Float64},
}
```
This makes LPs with `MOI.SingleVariable` conic programs, which is a bit overkill

**Comment by matbesancon** [comment]
this errors, because calls the conic interface:
```julia
    model = direct_model(diff_optimizer(Clp.Optimizer))
    MOI.set(model, MOI.Silent(), true)

    @variable(model, x[1:1])

    @objective(model, Min, x[1])
    @constraint(model, x[1] ≥ 0)
    @constraint(model, x[1] ≥ 3)

    optimize!(model)

    # obtain gradients
    grads = backward(JuMP.backend(model), ["G", "h"], [1.0])
```

This fixes the problem:
```julia
    @objective(model, Min, 1.1 * x[1])
```

**Comment by joaquimg** [comment]
This should be fixed in the #81 rewrite

**Comment by matbesancon** [comment]
OK great. I had a quick fix in an upcoming PR but a more systematic one will be better

**Comment by joaquimg** [comment]
fixed by #83
